tests/phpunit/tests/html-api/wpHtmlTagProcessorModifiableText.php

```diff
@@ -490,6 +490,7 @@ public static function data_unallowed_modifiable_text_updates() {
 			'Comment with --!>'                => array( '<!-- this is a comment -->', 'Invalid but legitimate comments end in --!>' ),
 			'SCRIPT with </script>'            => array( '<script>Replace me</script>', 'Just a </script>' ),
 			'SCRIPT with </script attributes>' => array( '<script>Replace me</script>', 'before</script id=sneak>after' ),
+			'SCRIPT with "<script " opener'    => array( '<script>Replace me</script>', '<!--<script ' ),
```

Comment:
earmarking for follow-up: we could probably create a new unit test that better spells out what we’re wanting to enshrine here

```php
/**
 * @param 'early-exit'|'prevents-exit' $violation Whether the violating contents close the SCRIPT element early or prevent its normal closure.
 */
public function rejects_script_contents_which_escape_the_script_element_boundaries( string $violating_script_contents, string $violation ) {
	…
}
```